diff --git a/node_modules/@react-native-community/netinfo/android/build.gradle b/node_modules/@react-native-community/netinfo/android/build.gradle
index 82b3c9a..d80f3e7 100644
--- a/node_modules/@react-native-community/netinfo/android/build.gradle
+++ b/node_modules/@react-native-community/netinfo/android/build.gradle
@@ -9,7 +9,7 @@ buildscript {
     }
 
     dependencies {
-      classpath("com.android.tools.build:gradle:4.2.2")
+      classpath("com.android.tools.build:gradle:7.3.1")
     }
   }
 }
@@ -29,7 +29,7 @@ def getExtOrIntegerDefault(name) {
 apply plugin: 'com.android.library'
 
 android {
-  compileSdkVersion getExtOrIntegerDefault('compileSdkVersion')
+  compileSdk getExtOrIntegerDefault('compileSdkVersion')
 
   compileOptions {
       sourceCompatibility JavaVersion.VERSION_1_8
@@ -56,7 +56,6 @@ repositories {
 }
 
 dependencies {
-  //noinspection GradleDynamicVersion
-  implementation 'com.facebook.react:react-native:+'
+  implementation 'com.facebook.react:react-native:0.68.1'
 
 }
diff --git a/node_modules/@react-native-community/netinfo/android/gradle.properties b/node_modules/@react-native-community/netinfo/android/gradle.properties
index e0a52d0..e44c0a2 100644
--- a/node_modules/@react-native-community/netinfo/android/gradle.properties
+++ b/node_modules/@react-native-community/netinfo/android/gradle.properties
@@ -1,4 +1,4 @@
-ReactNativeNetInfo_compileSdkVersion=29
-ReactNativeNetInfo_buildToolsVersion=29.0.3
-ReactNativeNetInfo_targetSdkVersion=27
-ReactNativeNetInfo_minSdkVersion=16
+ReactNativeNetInfo_compileSdkVersion=35
+ReactNativeNetInfo_buildToolsVersion=35.0.0
+ReactNativeNetInfo_targetSdkVersion=35
+ReactNativeNetInfo_minSdkVersion=23
