diff --git a/node_modules/@react-native-clipboard/clipboard/android/build.gradle b/node_modules/@react-native-clipboard/clipboard/android/build.gradle
index 86d4866..e33de37 100644
--- a/node_modules/@react-native-clipboard/clipboard/android/build.gradle
+++ b/node_modules/@react-native-clipboard/clipboard/android/build.gradle
@@ -5,7 +5,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.2.1'
+        classpath 'com.android.tools.build:gradle:7.3.1'
     }
 }
 
@@ -20,7 +20,7 @@ def getExtOrIntegerDefault(name) {
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion getExtOrIntegerDefault('compileSdkVersion')
+    compileSdk getExtOrIntegerDefault('compileSdkVersion')
 
     defaultConfig {
         minSdkVersion getExtOrIntegerDefault('minSdkVersion')
@@ -35,5 +35,5 @@ repositories {
 
 dependencies {
     //noinspection GradleDynamicVersion
-    implementation 'com.facebook.react:react-native:+'
+    implementation 'com.facebook.react:react-native:0.68.1'
 }
diff --git a/node_modules/@react-native-clipboard/clipboard/android/gradle.properties b/node_modules/@react-native-clipboard/clipboard/android/gradle.properties
index 936c1f0..a5ee705 100644
--- a/node_modules/@react-native-clipboard/clipboard/android/gradle.properties
+++ b/node_modules/@react-native-clipboard/clipboard/android/gradle.properties
@@ -1,4 +1,4 @@
-ReactNativeClipBoard_compileSdkVersion=30
-ReactNativeClipBoard_buildToolsVersion=30.0.2
-ReactNativeClipBoard_targetSdkVersion=30
-ReactNativeClipBoard_minSdkVersion=16
\ No newline at end of file
+ReactNativeClipBoard_compileSdkVersion=35
+ReactNativeClipBoard_buildToolsVersion=35.0.0
+ReactNativeClipBoard_targetSdkVersion=35
+ReactNativeClipBoard_minSdkVersion=23
