diff --git a/node_modules/react-native-yamap/RNYamap.podspec b/node_modules/react-native-yamap/RNYamap.podspec
index 571f8d2..e8cfb6f 100644
--- a/node_modules/react-native-yamap/RNYamap.podspec
+++ b/node_modules/react-native-yamap/RNYamap.podspec
@@ -8,11 +8,11 @@ Pod::Spec.new do |s|
     s.homepage     = "vvdev.ru"
     s.license      = "MIT"
     s.author       = { package["author"]["name"] => package["author"]["email"] }
-    s.platform     = :ios, "12.0"
+    s.platform     = :ios, "15.1"
     s.source       = { :git => "https://github.com/author/RNYamap.git", :tag => "master" }
     s.source_files = "ios/**/*.{h,m,swift}"
     # s.requires_arc = true
 
     s.dependency "React"
-    s.dependency "YandexMapsMobile", "4.8.0-full"
+    s.dependency "YandexMapsMobile", "4.19.0-full"
 end
diff --git a/node_modules/react-native-yamap/android/build.gradle b/node_modules/react-native-yamap/android/build.gradle
index d9de295..f6d365b 100644
--- a/node_modules/react-native-yamap/android/build.gradle
+++ b/node_modules/react-native-yamap/android/build.gradle
@@ -1,7 +1,7 @@
 
 buildscript {
     repositories {
-        jcenter()
+        google()
         mavenCentral()
     }
 }
@@ -12,9 +12,9 @@ apply plugin: 'org.jetbrains.kotlin.android'
 android {
 
     defaultConfig {
-        compileSdkVersion 34
-        minSdkVersion 26
-        targetSdkVersion 34
+        compileSdkVersion 36
+        minSdkVersion 24
+        targetSdkVersion 36
         versionCode 1
         versionName "1.1"
     }
@@ -31,8 +31,8 @@ repositories {
 }
 
 dependencies {
-    implementation 'com.google.android.gms:play-services-location:+'
-    implementation 'com.facebook.react:react-native:+'
-    implementation 'com.yandex.android:maps.mobile:4.8.0-full'
-    implementation 'androidx.core:core-ktx:1.13.1'
+    implementation 'com.google.android.gms:play-services-location:21.3.0'
+    implementation 'com.facebook.react:react-native:0.20.1'
+    implementation 'com.yandex.android:maps.mobile:4.19.0-full'
+    implementation 'androidx.core:core-ktx:1.16.0'
 }
diff --git a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/ClusteredYamapViewManager.kt b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/ClusteredYamapViewManager.kt
index 54d8a0f..b37dc38 100644
--- a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/ClusteredYamapViewManager.kt
+++ b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/ClusteredYamapViewManager.kt
@@ -128,7 +128,7 @@ class ClusteredYamapViewManager internal constructor() : ViewGroupManager<Cluste
             }
 
             "findRoutes" -> if (args != null) {
-                findRoutes(view, args.getArray(0), args.getArray(1), args.getString(2))
+                args.getString(2)?.let { findRoutes(view, args.getArray(0), args.getArray(1), it) }
             }
 
             "setZoom" -> if (args != null) {
@@ -152,11 +152,11 @@ class ClusteredYamapViewManager internal constructor() : ViewGroupManager<Cluste
             }
 
             "getScreenPoints" -> if (args != null) {
-                view.emitWorldToScreenPoints(args.getArray(0), args.getString(1))
+                args.getArray(0)?.let { view.emitWorldToScreenPoints(it, args.getString(1)) }
             }
 
             "getWorldPoints" -> if (args != null) {
-                view.emitScreenToWorldPoints(args.getArray(0), args.getString(1))
+                args.getArray(0)?.let { view.emitScreenToWorldPoints(it, args.getString(1)) }
             }
 
             else -> throw IllegalArgumentException(
@@ -243,7 +243,7 @@ class ClusteredYamapViewManager internal constructor() : ViewGroupManager<Cluste
             val vehicles = ArrayList<String>()
             if (jsVehicles != null) {
                 for (i in 0 until jsVehicles.size()) {
-                    vehicles.add(jsVehicles.getString(i))
+                    jsVehicles.getString(i)?.let { vehicles.add(it) }
                 }
             }
             castToYaMapView(view).findRoutes(points, vehicles, id)
diff --git a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapMarkerManager.kt b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapMarkerManager.kt
index e32d22c..17e7720 100644
--- a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapMarkerManager.kt
+++ b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapMarkerManager.kt
@@ -109,10 +109,12 @@ class YamapMarkerManager internal constructor() : ViewGroupManager<YamapMarker>(
             "animatedMoveTo" -> {
                 val markerPoint = args!!.getMap(0)
                 val moveDuration = args.getInt(1)
-                val lon = markerPoint.getDouble("lon").toFloat()
-                val lat = markerPoint.getDouble("lat").toFloat()
-                val point = Point(lat.toDouble(), lon.toDouble())
-                castToMarkerView(view).animatedMoveTo(point, moveDuration.toFloat())
+                val lon = markerPoint?.getDouble("lon")?.toFloat()
+                val lat = markerPoint?.getDouble("lat")?.toFloat()
+                val point = lon?.let { lat?.let { it1 -> Point(it1.toDouble(), it.toDouble()) } }
+                if (point != null) {
+                    castToMarkerView(view).animatedMoveTo(point, moveDuration.toFloat())
+                }
                 return
             }
 
diff --git a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapViewManager.kt b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapViewManager.kt
index 8b05905..7a5cc8e 100644
--- a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapViewManager.kt
+++ b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapViewManager.kt
@@ -128,7 +128,7 @@ class YamapViewManager internal constructor() : ViewGroupManager<YamapView>() {
             }
 
             "findRoutes" -> if (args != null) {
-                findRoutes(view, args.getArray(0), args.getArray(1), args.getString(2))
+                args.getString(2)?.let { findRoutes(view, args.getArray(0), args.getArray(1), it) }
             }
 
             "setZoom" -> if (args != null) {
@@ -152,11 +152,11 @@ class YamapViewManager internal constructor() : ViewGroupManager<YamapView>() {
             }
 
             "getScreenPoints" -> if (args != null) {
-                view.emitWorldToScreenPoints(args.getArray(0), args.getString(1))
+                args.getArray(0)?.let { view.emitWorldToScreenPoints(it, args.getString(1)) }
             }
 
             "getWorldPoints" -> if (args != null) {
-                view.emitScreenToWorldPoints(args.getArray(0), args.getString(1))
+                args.getArray(0)?.let { view.emitScreenToWorldPoints(it, args.getString(1)) }
             }
 
             else -> throw IllegalArgumentException(
@@ -237,7 +237,7 @@ class YamapViewManager internal constructor() : ViewGroupManager<YamapView>() {
 
             if (jsVehicles != null) {
                 for (i in 0 until jsVehicles.size()) {
-                    vehicles.add(jsVehicles.getString(i))
+                    jsVehicles.getString(i)?.let { vehicles.add(it) }
                 }
             }
 
diff --git a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/view/ClusteredYamapView.kt b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/view/ClusteredYamapView.kt
index c01f591..c492365 100644
--- a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/view/ClusteredYamapView.kt
+++ b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/view/ClusteredYamapView.kt
@@ -23,7 +23,7 @@ class ClusteredYamapView(context: Context?) : YamapView(context), ClusterListene
     private val placemarksMap: HashMap<String?, PlacemarkMapObject?> = HashMap<String?, PlacemarkMapObject?>()
     private var pointsList = ArrayList<Point>()
 
-    fun setClusteredMarkers(points: ArrayList<Any>) {
+    fun setClusteredMarkers(points: java.util.ArrayList<Any?>) {
         clusterCollection.clear()
         placemarksMap.clear()
         val pt = ArrayList<Point>()
diff --git a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/view/YamapView.kt b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/view/YamapView.kt
index 48c1cdf..a917265 100644
--- a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/view/YamapView.kt
+++ b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/view/YamapView.kt
@@ -84,7 +84,7 @@ open class YamapView(context: Context?) : MapView(context), UserLocationObjectLi
     private var userLocationIconScale = 1f
     private var userLocationBitmap: Bitmap? = null
     private val routeMng = RouteManager()
-    private var routeOptions: RouteOptions = RouteOptions(FitnessOptions(false))
+    private var routeOptions: RouteOptions = RouteOptions(FitnessOptions(false, false))
     private val masstransitRouter = TransportFactory.getInstance().createMasstransitRouter()
     private val drivingRouter: DrivingRouter
     private val pedestrianRouter = TransportFactory.getInstance().createPedestrianRouter()
@@ -226,8 +226,8 @@ open class YamapView(context: Context?) : MapView(context), UserLocationObjectLi
 
         for (i in 0 until worldPoints.size()) {
             val p = worldPoints.getMap(i)
-            val worldPoint = Point(p.getDouble("lat"), p.getDouble("lon"))
-            val screenPoint = mapWindow.worldToScreen(worldPoint)
+            val worldPoint = p?.let { Point(p.getDouble("lat"), it.getDouble("lon")) }
+            val screenPoint = worldPoint?.let { mapWindow.worldToScreen(it) }
             screenPoints.pushMap(screenPointToJSON(screenPoint))
         }
 
@@ -245,8 +245,9 @@ open class YamapView(context: Context?) : MapView(context), UserLocationObjectLi
 
         for (i in 0 until screenPoints.size()) {
             val p = screenPoints.getMap(i)
-            val screenPoint = ScreenPoint(p.getDouble("x").toFloat(), p.getDouble("y").toFloat())
-            val worldPoint = mapWindow.screenToWorld(screenPoint)
+            val screenPoint = p?.getDouble("x")
+                ?.let { ScreenPoint(it.toFloat(), p.getDouble("y").toFloat()) }
+            val worldPoint = screenPoint?.let { mapWindow.screenToWorld(it) }
             worldPoints.pushMap(worldPointToJSON(worldPoint))
         }
 
@@ -295,7 +296,7 @@ open class YamapView(context: Context?) : MapView(context), UserLocationObjectLi
             val _points = ArrayList<RequestPoint>()
             for (i in points.indices) {
                 val point = points[i]
-                val _p = RequestPoint(point!!, RequestPointType.WAYPOINT, null, null)
+                val _p = RequestPoint(point!!, RequestPointType.WAYPOINT, null, null, null)
                 _points.add(_p)
             }
 
@@ -310,7 +311,7 @@ open class YamapView(context: Context?) : MapView(context), UserLocationObjectLi
         val _points = ArrayList<RequestPoint>()
         for (i in points.indices) {
             val point = points[i]
-            _points.add(RequestPoint(point!!, RequestPointType.WAYPOINT, null, null))
+            _points.add(RequestPoint(point!!, RequestPointType.WAYPOINT, null, null, null))
         }
         val listener: Session.RouteListener = object : Session.RouteListener {
             override fun onMasstransitRoutes(routes: List<Route>) {
@@ -590,10 +591,12 @@ open class YamapView(context: Context?) : MapView(context), UserLocationObjectLi
         if (show) {
             userLocationLayer!!.setObjectListener(this)
             userLocationLayer!!.isVisible = true
-            userLocationLayer!!.isHeadingEnabled = true
+            userLocationLayer!!.isHeadingModeActive = true
+            // userLocationLayer!!.isHeadingEnabled = true
         } else {
             userLocationLayer!!.isVisible = false
-            userLocationLayer!!.isHeadingEnabled = false
+            userLocationLayer!!.isHeadingModeActive = false
+            // userLocationLayer!!.isHeadingEnabled = false
             userLocationLayer!!.setObjectListener(null)
         }
     }
@@ -685,7 +688,7 @@ open class YamapView(context: Context?) : MapView(context), UserLocationObjectLi
         val wTransports = Arguments.createMap()
 
         for ((key, value) in transports) {
-            wTransports.putArray(key, Arguments.fromList(value))
+            wTransports.putArray(key, value?.let { Arguments.fromList(it) })
         }
 
         routeMetadata.putMap("transports", wTransports)
diff --git a/node_modules/react-native-yamap/ios/ClusteredYamapView.m b/node_modules/react-native-yamap/ios/ClusteredYamapView.m
index ce9fd1e..c3ccb15 100644
--- a/node_modules/react-native-yamap/ios/ClusteredYamapView.m
+++ b/node_modules/react-native-yamap/ios/ClusteredYamapView.m
@@ -186,7 +186,7 @@ - (void)setCenterForMap:(RNCYMView*)map center:(NSDictionary*)_center zoom:(floa
         NSArray<YMKPoint*>* points = [RCTConvert Points:json[@"points"]];
         NSMutableArray<YMKRequestPoint*>* requestPoints = [[NSMutableArray alloc] init];
         for (int i = 0; i < [points count]; ++i) {
-            YMKRequestPoint * requestPoint = [YMKRequestPoint requestPointWithPoint:[points objectAtIndex:i] type: YMKRequestPointTypeWaypoint pointContext:nil drivingArrivalPointId:nil];
+            YMKRequestPoint * requestPoint = [YMKRequestPoint requestPointWithPoint:[points objectAtIndex:i] type: YMKRequestPointTypeWaypoint pointContext:nil drivingArrivalPointId:nil indoorLevelId:nil];
             [requestPoints addObject:requestPoint];
         }
         NSArray<NSString*>* vehicles = [RCTConvert Vehicles:json[@"vehicles"]];
diff --git a/node_modules/react-native-yamap/ios/View/RNYMView.m b/node_modules/react-native-yamap/ios/View/RNYMView.m
index 70691f8..9febc6b 100644
--- a/node_modules/react-native-yamap/ios/View/RNYMView.m
+++ b/node_modules/react-native-yamap/ios/View/RNYMView.m
@@ -64,7 +64,7 @@ - (instancetype)init {
     drivingRouter = [[YMKDirectionsFactory instance] createDrivingRouterWithType:YMKDrivingRouterTypeOnline];
     pedestrianRouter = [[YMKTransportFactory instance] createPedestrianRouter];
     transitOptions = [YMKTransitOptions transitOptionsWithAvoid:YMKFilterVehicleTypesNone timeOptions:[[YMKTimeOptions alloc] init]];    acceptVehicleTypes = [[NSMutableArray<NSString *> alloc] init];
-    routeOptions = [YMKRouteOptions routeOptionsWithFitnessOptions:[YMKFitnessOptions fitnessOptionsWithAvoidSteep:false]];
+    routeOptions = [YMKRouteOptions routeOptionsWithFitnessOptions:[YMKFitnessOptions fitnessOptionsWithAvoidSteep:false avoidStairs:false ]];
     routes = [[NSMutableArray alloc] init];
     currentRouteInfo = [[NSMutableArray alloc] init];
     lastKnownRoutePoints = [[NSMutableArray alloc] init];
diff --git a/node_modules/react-native-yamap/ios/YamapView.m b/node_modules/react-native-yamap/ios/YamapView.m
index d9ebae4..e4ddc1b 100644
--- a/node_modules/react-native-yamap/ios/YamapView.m
+++ b/node_modules/react-native-yamap/ios/YamapView.m
@@ -212,7 +212,7 @@ - (void)setCenterForMap:(RNYMView*)map center:(NSDictionary*)_center zoom:(float
         NSMutableArray<YMKRequestPoint *> *requestPoints = [[NSMutableArray alloc] init];
 
         for (int i = 0; i < [points count]; ++i) {
-            YMKRequestPoint *requestPoint = [YMKRequestPoint requestPointWithPoint:[points objectAtIndex:i] type:YMKRequestPointTypeWaypoint pointContext:nil drivingArrivalPointId:nil];
+            YMKRequestPoint *requestPoint = [YMKRequestPoint requestPointWithPoint:[points objectAtIndex:i] type:YMKRequestPointTypeWaypoint pointContext:nil drivingArrivalPointId:nil indoorLevelId:nil];
             [requestPoints addObject:requestPoint];
         }
 
