diff --git a/node_modules/react-native-change-icon/android/build.gradle b/node_modules/react-native-change-icon/android/build.gradle
index a078aaa..8a2be70 100644
--- a/node_modules/react-native-change-icon/android/build.gradle
+++ b/node_modules/react-native-change-icon/android/build.gradle
@@ -1,3 +1,5 @@
+import com.android.Version
+
 buildscript {
   repositories {
     google()
@@ -5,7 +7,7 @@ buildscript {
   }
 
   dependencies {
-    classpath "com.android.tools.build:gradle:7.2.1"
+    classpath "com.android.tools.build:gradle:8.13.2"
   }
 }
 
@@ -31,7 +33,7 @@ def getExtOrIntegerDefault(name) {
 }
 
 def supportsNamespace() {
-  def parsed = com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION.tokenize('.')
+  def parsed = Version.ANDROID_GRADLE_PLUGIN_VERSION.tokenize('.')
   def major = parsed[0].toInteger()
   def minor = parsed[1].toInteger()
 
@@ -72,8 +74,8 @@ android {
   }
 
   compileOptions {
-    sourceCompatibility JavaVersion.VERSION_1_8
-    targetCompatibility JavaVersion.VERSION_1_8
+    sourceCompatibility JavaVersion.VERSION_17
+    targetCompatibility JavaVersion.VERSION_17
   }
 
 }
diff --git a/node_modules/react-native-change-icon/android/src/main/java/com/reactnativechangeicon/ChangeIconModule.java b/node_modules/react-native-change-icon/android/src/main/java/com/reactnativechangeicon/ChangeIconModule.java
index b303063..cb77796 100644
--- a/node_modules/react-native-change-icon/android/src/main/java/com/reactnativechangeicon/ChangeIconModule.java
+++ b/node_modules/react-native-change-icon/android/src/main/java/com/reactnativechangeicon/ChangeIconModule.java
@@ -62,11 +62,12 @@ public class ChangeIconModule extends ReactContextBaseJavaModule implements Appl
     @ReactMethod
     public void changeIcon(String iconName, Promise promise) {
         final Activity activity = getCurrentActivity();
-        final String activityName = activity.getComponentName().getClassName();
         if (activity == null) {
             promise.reject("ANDROID:ACTIVITY_NOT_FOUND");
             return;
         }
+
+        final String activityName = activity.getComponentName().getClassName();
         if (this.componentClass.isEmpty()) {
             this.componentClass = activityName.endsWith("MainActivity") ? activityName + "Default" : activityName;
         }
@@ -109,31 +110,31 @@ public class ChangeIconModule extends ReactContextBaseJavaModule implements Appl
     }
 
     @Override
-    public void onActivityPaused(Activity activity) {
+    public void onActivityPaused(@NonNull Activity activity) {
         completeIconChange();
     }
 
     @Override
-    public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
+    public void onActivityCreated(@NonNull Activity activity, Bundle savedInstanceState) {
     }
 
     @Override
-    public void onActivityStarted(Activity activity) {
+    public void onActivityStarted(@NonNull Activity activity) {
     }
 
     @Override
-    public void onActivityResumed(Activity activity) {
+    public void onActivityResumed(@NonNull Activity activity) {
     }
 
     @Override
-    public void onActivityStopped(Activity activity) {
+    public void onActivityStopped(@NonNull Activity activity) {
     }
 
     @Override
-    public void onActivitySaveInstanceState(Activity activity, Bundle outState) {
+    public void onActivitySaveInstanceState(@NonNull Activity activity, @NonNull Bundle outState) {
     }
 
     @Override
-    public void onActivityDestroyed(Activity activity) {
+    public void onActivityDestroyed(@NonNull Activity activity) {
     }
 }
diff --git a/node_modules/react-native-change-icon/ios/ChangeIcon.h b/node_modules/react-native-change-icon/ios/ChangeIcon.h
index 3624fe4..3547915 100644
--- a/node_modules/react-native-change-icon/ios/ChangeIcon.h
+++ b/node_modules/react-native-change-icon/ios/ChangeIcon.h
@@ -1,11 +1,6 @@
 #import <UIKit/UIKit.h>
 
-#ifdef RCT_NEW_ARCH_ENABLED
-#import "RNChangeIconSpec.h"
-@interface ChangeIcon : NSObject <NativeChangeIconSpec, UIApplicationDelegate>
-#else
 #import <React/RCTBridgeModule.h>
 @interface ChangeIcon : NSObject <RCTBridgeModule, UIApplicationDelegate>
-#endif
 
 @end
